allen-cell-animated/cell-feature-data
# Python dataset processor: create and write files

Problem
=======
What is the problem this work solves, including
#94 

Solution
========

Implemented the `UserInputHandler` class to gather required settings from user inputs, and the `DatasetWriter` class to process datasets and write json files. Key steps include:

- [x] get and validate basic user inputs (`version`, `title`, etc )
- [x] create a new folder within `/data`
- [x] separate dataset info into designed json files
- [x] process each row and column to extract and validate data we need for compiling each json file

To-dos after this:
https://github.com/allen-cell-animated/cell-feature-data/issues/97
https://github.com/allen-cell-animated/cell-feature-data/issues/98

## Type of change
Please delete options that are not relevant.

* New feature (non-breaking change which adds functionality)


Steps to Verify:
----------------
1. run `npm run create-dataset -- dataset-processor-python/tests/test_data.csv`
2. for testing, input the version (e.g. `2023.11`), and leave the `title` and `description` fields empty

<img width="307" alt="Screenshot 2023-11-27 at 4 59 08 PM" src="https://github.com/allen-cell-animated/cell-feature-data/assets/91452427/8038e3cc-c925-4fa1-8df1-c4b7de9c41df">

4. confirm a folder named `test_data_v2023.11` has been created within `/data`, containing the following file structure:
 
<img width="402" alt="Screenshot 2023-11-27 at 4 48 12 PM" src="https://github.com/allen-cell-animated/cell-feature-data/assets/91452427/62ab9c65-fbbb-4986-b806-955e095e7007">

5. review each json file. Note: `image_settings.json`is currently blank. Some keys in `options` within `feature_defs` need further improvements in the next subsequent issues



Keyfiles (delete if not relevant):
-----------------------
1. `data_loader.py`
